docs/model_access_control.md

```diff
@@ -0,0 +1,652 @@
+
+```
+PUT /_cluster/settings
+{
+  "persistent" : {
+    "plugins.ml_commons.model_access_control_enabled" : false 
+  }
+}
+```
+
+For the purpose of this step-by-step tutorial, let us consider admin and four other users with "ml_full_access" permission role i.e. all the users have permissions to perform all actions on ml resources. For more information on users and roles, refer to https://opensearch.org/docs/latest/security/access-control/users-roles/
+
+| Users   | Backend role                      |
+|:--------|:----------------------------------|
+| `User1` | IT, HR                            |
+| `User2` | IT                                |
+| `User3` | Finance                           |
+| `User4` | -                                 |
+| `Admin` | Admin can assign any backend role |
+
+
+
+## Model groups
+
+Users can register different versions to the same model. For ease of control, these versions can be registered under a same model group. So, user first needs to create a model group and then start registering model versions to it. A model group can have any number of versions associated with it.
+
+The access control in ml-commons is implemented at the _model group level_ so that all versions of the same model share the same access mode as that of their model group. That is, if user has access to one version, then he/she should have access to all versions of the model group.
+
+When creating a model group, user should also specify the access mode to it. Depending on the access mode, other users will be able to access the versions of a model group or create new versions to it.
+
+If a user tries to register/deploy/undeploy/predict/delete/search/get model versions, first we check if the user has access to the model. Otherwise, we throw exception saying user does not have access to the model specified.
+
+A user is considered a model _owner_ when he/she creates a new model group and register its first version. When a model owner creates a model group, the owner can specify one of the following _access modes_ for this model group:
+
+- `public`: All users who have access to the cluster can access this model group.
+- `private`: Only the model owner or an admin user can access this model group.
+- `restricted`: The owner, an admin user, or any user who shares one of the model group's backend roles can access any model in this model group. When creating a `restricted` model group, the owner must attach one or more of the owner's backend roles to the model. For a user to be able to access a model group, one of the user's backend roles must match one of the model group's backend roles. This grants the user permissions associated with the user role to which that backend role is mapped. For example, if the backend role is mapped to the `ml_full_access` user role, users with this role can use the Register, Deploy, Predict, Delete, Search and Get Model APIs on any model in this model group.
```

Comment:
"This grants the user permissions associated with the user role to which that backend role is mapped. For example, if the backend role is mapped to the `ml_full_access` user role, users with this role can use the Register, Deploy, Predict, Delete, Search and Get Model APIs on any model in this model group."

Seems not related with `restricted`. Remove these or move to other part?